fission/src/mirabuf/MirabufSceneObject.ts

```diff
@@ -77,11 +106,35 @@ class MirabufSceneObject extends SceneObject {
                     .clone()
                     .premultiply(transform)
                 this._mirabufInstance.meshes.get(part)!.forEach(mesh => {
+                    // iterating through each mesh and updating their position and rotation
                     mesh.position.setFromMatrixPosition(partTransform)
                     mesh.rotation.setFromRotationMatrix(partTransform)
                 })
             })
 
+            /**
+             * transform gizmo block to update the position and rotation of the body to match the gizmos's position
+             */
+            if (this.transformGizmos.isActive) {
+                // commands to either cancel gizmo creation or confirm position
+                if (InputSystem.getInput("enter")) {
+                    this.transformGizmos.RemoveGizmos()
+                    World.PhysicsSystem.EnablePhysicsForBody(this._mechanism.GetBodyByNodeId(rn.id)!)
+                    World.PhysicsSystem.EnablePhysicsForBody(this._mechanism.GetBodyByNodeId(this._mechanism.rootBody)!)
+                    return
+                } else if (InputSystem.getInput("escape")) {
+                    this.transformGizmos.RemoveGizmos()
+                    World.SceneRenderer.RemoveSceneObject(this.id)
+                    return
+                }
+
+                // if the gizmo is being dragged, copy the mesh position and rotation to the Mirabuf body
+                if (this.transformGizmos.isBeingDragged()) {
+                    this.transformGizmos.UpdateMirabufPositioning(this, rn)
+                    World.PhysicsSystem.DisablePhysicsForBody(this._mechanism.GetBodyByNodeId(rn.id)!)
+                }
+            }
```

Comment:
You should move this to only be done once per update, not for each node. You're running into an issue where enter and escape are only being detected in one of these loops before being discarded so running enable on all the rigid nodes doesn't work.